contracts/swapper/LevSwapper/curve/CurveLevSwapper2Tokens.sol

```diff
@@ -46,21 +39,22 @@ abstract contract CurveLevSwapper2Tokens is BaseLevSwapper {
     }
 
     /// @inheritdoc BaseLevSwapper
-    function _remove(uint256 burnAmount, bytes memory data) internal override returns (uint256 amountOut) {
+    function _remove(uint256 burnAmount, bytes memory data) internal override {
         CurveRemovalType removalType;
         (removalType, data) = abi.decode(data, (CurveRemovalType, bytes));
         if (removalType == CurveRemovalType.oneCoin) {
             (int128 whichCoin, uint256 minAmountOut) = abi.decode(data, (int128, uint256));
-            amountOut = metapool().remove_liquidity_one_coin(burnAmount, whichCoin, minAmountOut);
+            metapool().remove_liquidity_one_coin(burnAmount, whichCoin, minAmountOut);
         } else if (removalType == CurveRemovalType.balance) {
             uint256[2] memory minAmountOuts = abi.decode(data, (uint256[2]));
-            minAmountOuts = metapool().remove_liquidity(burnAmount, minAmountOuts);
+            metapool().remove_liquidity(burnAmount, minAmountOuts);
         } else if (removalType == CurveRemovalType.imbalance) {
             (address to, uint256[2] memory amountOuts) = abi.decode(data, (address, uint256[2]));
-            uint256 actualBurnAmount = metapool().remove_liquidity_imbalance(amountOuts, burnAmount);
+            metapool().remove_liquidity_imbalance(amountOuts, burnAmount);
+            uint256 keptAmount = lpToken().balanceOf(address(this));
             // We may have withdrawn more than needed: maybe not optimal because a user may not want to have
             // lp tokens staked. Solution is to do a sweep on all tokens in the `BaseLevSwapper` contract
-            if (burnAmount > actualBurnAmount) angleStaker().deposit(burnAmount - actualBurnAmount, to);
+            if (keptAmount > 0) angleStaker().deposit(keptAmount, to);
```

Comment:
why do we deposit everything compared to previously where we deposited only the surplus ?

Reply:
This is the same amount, just the previous interface returned the amount burnt (so we had left what has been withdraw minus what was burnt). Now we just check the left amount (which is equal to the past computation)